src/semantics/UHExp.re

```diff
@@ -111,78 +122,124 @@ let bidelimited = (e: t): bool =>
   | Parenthesized(_) => true
   /* non-bidelimited cases */
   | Tm(_, Case(_, _, _))
-  | Tm(_, LineItem(_, _))
   | Tm(_, Lam(_, _, _))
-  | Tm(_, OpSeq(_, _)) => false
+  | OpSeq(_, _) => false
   };
 
 /* if e is not bidelimited, bidelimit e parenthesizes it */
 let bidelimit = (e: t): t =>
   if (bidelimited(e)) {
     e;
   } else {
-    Parenthesized(e);
+    Parenthesized(Block([], e));
+  };
+
+let rec get_err_status_block = (Block(_, e): block): err_status =>
+  get_err_status_t(e)
```

Comment:
My first attempt had err_status at the block level, but then I removed it because I didn't want to deal with figuring out how to localize type errors (is the type error on the block or the concluding expression?). I think there were other reasons too that I'm forgetting now. As long as we don't have a err_status at the block level, it seems like we should just rely on the concluding expression